bundles/org.openhab.ui/web/src/pages/settings/persistence/persistence-edit.vue

```diff
@@ -539,16 +674,32 @@ export default {
       this.checkDirty()
     },
     deleteModule (ev, module, index) {
+      if (!this.editable) return
       let swipeoutElement = ev.target
+      const self = this
+      ev.cancelBubble = true
+      while (!swipeoutElement.classList.contains('swipeout')) {
+        swipeoutElement = swipeoutElement.parentElement
+      }
+      this.$f7.swipeout.delete(swipeoutElement, () => {
+        console.debug(`Removing ${module}:`)
+        console.debug(self.persistence[module][index])
+        self.persistence[module].splice(index, 1)
+        this.checkDirty()
+      })
+    },
+    deleteModuleKey (ev, module, key) {
       if (!this.editable) return
+      let swipeoutElement = ev.target
+      const self = this
```

Comment:
Normally there is no need for `self = this` "hacks" when using array functions in callbacks, which is the case here.
Why did you use this "hack" here?